### PR TITLE
Thing model types

### PR DESCRIPTION
**Pull request by relu91**
As discussed in the previous calls, this PR adds thing models types to our repository. It follows the same logic of the thing-description package with automatic synchronization to [ wot-thing-description](https://github.com/w3c/wot-thing-description). 

fixes https://github.com/w3c/wot-scripting-api/pull/373

### CODE REVIEW

**Review by danielpeintner — CHANGES_REQUESTED** [review]
LGTM

Minor:
I suggest to add a link to the main TS readme
https://github.com/w3c/wot-scripting-api/tree/main/typescript

File: typescript/thing-model/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# wot-thing-description-types
```
Comment (danielpeintner) [inline typescript/thing-model/README.md:1]: name needs to be fixed

**Review by danielpeintner — APPROVED** [review]

**Comment by danielpeintner** [comment]
Scripting API 2022-01-31

* @relu91 fixes git history. Can be merged after that.

**Review by JKRhb — APPROVED** [review]

**Comment by relu91** [comment]
@danielpeintner ready to go (I've just fixed a rebase leftover)

**Comment by danielpeintner** [comment]
@relu91 my comment about readme was more about the main readme were we currently have 2 items but with this PR it should be 3

grafik

**Comment by relu91** [comment]
@danielpeintner ✅

**Pull request merged**